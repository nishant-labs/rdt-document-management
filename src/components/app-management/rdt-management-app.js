import React, { Component } from 'react';
import Box from '@material-ui/core/Box';
import { SearchBox } from '../search-box';
import { AppLayout } from '../../layout/app-container';
import { ThemeProvider } from '../../layout/theme-provider';
import { StatisticsTable } from '../statistics-table';
import { TypographyWithBox } from '../TypographyWithBox';
import { withConsumers } from '../../hoc/withConsumers';

export class RDTAppManagement extends Component {
	render() {
		const {
			populationData,
			gdpData,
			populationApiError,
			gdpApiError,
		} = this.props;
		const hasErrors = populationApiError || gdpApiError;
		const isDataNotAvailable = !hasErrors && (!populationData || !gdpData);
		return (
			<ThemeProvider>
				<AppLayout>
					<TypographyWithBox
						type="h1"
						text="RDT Document Management Communicate"
					/>
					<SearchBox />
					{hasErrors && (
						<TypographyWithBox
							type="body1"
							text="Something went wrong while fetching data"
						/>
					)}
					{isDataNotAvailable && (
						<TypographyWithBox
							type="body1"
							text="Use search box to get Population and GDP statistics from world bank for a specific year."
						/>
					)}
					{!isDataNotAvailable && !hasErrors && (
						<Box
							p={1}
							display="flex"
							flexDirection="row"
							justifyContent="center"
						>
							<StatisticsTable population={populationData} gdp={gdpData} />
						</Box>
					)}
				</AppLayout>
			</ThemeProvider>
		);
	}
}

export default withConsumers(RDTAppManagement);
