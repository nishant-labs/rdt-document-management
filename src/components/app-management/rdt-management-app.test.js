import { shallow } from 'enzyme';
import { RDTAppManagement } from './rdt-management-app';

const setupTest = (setupProps = {}) => {
  const baseProps = {
    populationData: {
      countryId: {
        countryName: 'countryName',
        population: 1234,
      },
    },
    gdpData: {
      countryId: {
        countryName: 'countryName',
        gdp: 12,
      },
    },
    populationApiError: null,
    gdpApiError: null,
  };
  const props = { ...baseProps, ...setupProps };
  const wrapper = shallow(<RDTAppManagement {...props} />);
  return {
    wrapper,
    props,
  };
};

describe('components/app-management/rdt-management-app', () => {
  test('should render with correct data', () => {
    const { wrapper } = setupTest();
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should pass correct data to table', () => {
    const { wrapper, props } = setupTest();
    const tableWrapper = wrapper.find('GridTableContainer');
    expect(tableWrapper.prop('population')).toEqual(props.populationData);
    expect(tableWrapper.prop('gdp')).toEqual(props.gdpData);
  });

  test('should render header', () => {
    const { wrapper } = setupTest();
    const header = wrapper.find('TypographyWithBox');
    expect(header.prop('type')).toEqual('h1');
    expect(header.prop('text')).toEqual('RDT Document Management Communicate');
  });

  test('should render no data message', () => {
    const { wrapper } = setupTest({ populationData: null, gdpData: null });
    const noMessageWrapper = wrapper.find('TypographyWithBox').at(1);
    expect(noMessageWrapper.prop('type')).toEqual('body1');
    expect(noMessageWrapper.prop('text')).toEqual(
      'Use search box to get Population and GDP statistics from world bank for a specific year.'
    );
  });

  test('should render error message', () => {
    const { wrapper } = setupTest({
      populationData: null,
      gdpData: null,
      populationApiError: { message: 'error' },
    });
    const errorWrapper = wrapper.find('TypographyWithBox').at(1);
    expect(errorWrapper.prop('type')).toEqual('body1');
    expect(errorWrapper.prop('text')).toEqual(
      'Something went wrong while fetching data'
    );
  });
});
