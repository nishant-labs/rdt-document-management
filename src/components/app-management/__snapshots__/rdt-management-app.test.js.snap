// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/app-management/rdt-management-app should render with correct data 1`] = `
"<AppThemeProvider>
  <AppLayoutContainer>
    <TypographyWithBox type=\\"h1\\" text=\\"RDT Document Management Communicate\\" />
    <SearchBoxView />
    <Styled(MuiBox) p={1} display=\\"flex\\" flexDirection=\\"row\\" justifyContent=\\"center\\">
      <GridTableContainer population={{...}} gdp={{...}} />
    </Styled(MuiBox)>
  </AppLayoutContainer>
</AppThemeProvider>"
`;
